static/app/views/settings/components/settingsLayout.tsx

```diff
@@ -17,100 +17,72 @@ type Props = {
+  const headerRef = useRef<HTMLDivElement>(null);
 
-  toggleNav(navVisible: boolean) {
-    // when the navigation is opened, body should be scroll-locked
-    this.toggleBodyScrollLock(navVisible);
+  function toggleNav(visible: boolean) {
+    const bodyElement = document.getElementsByTagName('body')[0];
 
-    this.setState({
-      navOffsetTop: this.headerRef.current?.getBoundingClientRect().bottom ?? 0,
-      navVisible,
-    });
+    window.scrollTo?.(0, 0);
+    bodyElement.classList[visible ? 'add' : 'remove']('scroll-lock');
+
+    setNavVisible(visible);
+    setNavOffsetTop(headerRef.current?.getBoundingClientRect().bottom ?? 0);
   }
 
-  toggleBodyScrollLock(lock: boolean) {
-    const bodyElement = document.getElementsByTagName('body')[0];
+  // Close menu when navigating away
+  useEffect(() => browserHistory.listen(() => toggleNav(false)), []);
 
-    if (window.scrollTo) {
-      window.scrollTo(0, 0);
-    }
-    bodyElement.classList[lock ? 'add' : 'remove']('scroll-lock');
-  }
+  const {renderNavigation, children, params, routes, route} = props;
 
-  render() {
-    const {params, routes, route, renderNavigation, children} = this.props;
-    const {navVisible, navOffsetTop} = this.state;
-
-    // We want child's view's props
-    const childProps =
-      children && React.isValidElement(children) ? children.props : this.props;
-    const childRoutes = childProps.routes || routes || [];
-    const childRoute = childProps.route || route || {};
-    const shouldRenderNavigation = typeof renderNavigation === 'function';
-
-    return (
-      <SettingsColumn>
-        <SettingsHeader ref={this.headerRef}>
-          <HeaderContent>
-            {shouldRenderNavigation && (
-              <NavMenuToggle
-                priority="link"
-                label={t('Open the menu')}
-                icon={navVisible ? <IconClose aria-hidden /> : <IconMenu aria-hidden />}
-                onClick={() => this.toggleNav(!navVisible)}
-              />
-            )}
-            <StyledSettingsBreadcrumb
-              params={params}
-              routes={childRoutes}
-              route={childRoute}
-            />
-            <SettingsSearch />
-          </HeaderContent>
-        </SettingsHeader>
+  // We want child's view's props
+  const childProps = children && isValidElement(children) ? children.props : props;
+  const childRoutes = childProps.routes || routes || [];
+  const childRoute = childProps.route || route || {};
+  const shouldRenderNavigation = typeof renderNavigation === 'function';
 
-        <MaxWidthContainer>
+  return (
+    <SettingsColumn>
+      <SettingsHeader ref={headerRef}>
+        <HeaderContent>
           {shouldRenderNavigation && (
-            <SidebarWrapper isVisible={navVisible} offsetTop={navOffsetTop}>
-              {renderNavigation!()}
-            </SidebarWrapper>
+            <NavMenuToggle
+              priority="link"
+              label={t('Open the menu')}
```

Comment:
Feature creeping here, but:
```suggestion
              label={navVisible ? t('Close the menu') : t('Open the menu')}
```

Reply:
will resolve in a another PR